blockchain/versionbits.go

```diff
@@ -123,27 +120,36 @@ type deploymentChecker struct {
 // interface.
 var _ thresholdConditionChecker = deploymentChecker{}
 
-// BeginTime returns the unix timestamp for the median block time after which
-// voting on a rule change starts (at the next window).
+// HasEnded returns true if the target consensus rule change has expired
+// or timed out (at the next window).
 //
 // This implementation returns the value defined by the specific deployment the
 // checker is associated with.
 //
 // This is part of the thresholdConditionChecker interface implementation.
-func (c deploymentChecker) BeginTime() uint64 {
-	return c.deployment.StartTime
+func (c deploymentChecker) HasStarted(blkNode *blockNode) bool {
+	// Can't fail as we make sure to set the clock above when we
+	// instantiate *BlockChain.
+	header := blkNode.Header()
+	started, _ := c.deployment.DeploymentStarter.HasStarted(&header)
```

Comment:
I realize it can't fail as is right now, but we should handle the error anyway in case this is not caught/updated whenever a new fallible deployment implementation is added.

Reply:
As is, starting can never fail. When/if a deployment that can fail here (which may come in a few years given the current rate of soft forks in the system), we can thread through the error. With the current function signature, not having to check the error makes the `thresholdStateTransition` more readable as there's less line noise.